src/modulefinder/sentry_modulefinder_linux.c

```diff
@@ -41,6 +43,50 @@ static sentry_value_t g_modules = { 0 };
 
 static sentry_slice_t LINUX_GATE = { "linux-gate.so", 13 };
 
+bool
+sentry__mmap_file(sentry_mmap_t *rv, const char *path)
+{
+    int fd = open(path, O_RDONLY);
+    if (fd < 0) {
+        goto fail;
+    }
+
+    struct stat sb;
+    if (stat(path, &sb) != 0 || !S_ISREG(sb.st_mode)) {
+        goto fail;
+    }
+
+    rv->len = sb.st_size;
+    if (rv->len == 0) {
+        goto fail;
+    }
+
+    rv->ptr = mmap(NULL, rv->len, PROT_READ, MAP_PRIVATE, fd, 0);
+    if (rv->ptr == MAP_FAILED) {
+        goto fail;
+    }
+
+    return true;
```

Comment:
the fd currently leaks in the success path, you still need to close it yourself.

Reply:
nice catch!